pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -654,7 +654,10 @@ let
 
       MODULE_COMPRESS    = yes;
       MODULE_COMPRESS_XZ = yes;
-      KERNEL_XZ          = yes;
+
+      # use zstd for kernel compression if newer than 5.9, else xz.
+      KERNEL_XZ          = { optional = true; tristate = whenOlder "5.9" "y";};
+      KERNEL_ZSTD        = { optional = true; tristate = whenAtLeast "5.9" "y";};
```

Comment:
I don't think `optional` is necessary here, since the version check already ensures that it's available, and we probably have some other issue if it's still not available.